### PR TITLE
Support for Streaming server metrics

### PR DESCRIPTION
**Pull request by wallyqs**
Picks up from #54 and adds support for polling streaming server metrics from serverz and channelsz endpoints.

```
go run prometheus_nats_exporter.go -channelz -serverz http://localhost:8222
```

Updates the build and dependencies as well.

Example:

```sh
# HELP nss_chan_bytes_total Total of bytes
# TYPE nss_chan_bytes_total counter
nss_chan_bytes_total{channel="foo",server="http://localhost:8222"} 48

# HELP nss_chan_last_seq Last seq
# TYPE nss_chan_last_seq gauge
nss_chan_last_seq{channel="foo",server="http://localhost:8222"} 2

# HELP nss_chan_msgs_total Total of messages
# TYPE nss_chan_msgs_total counter
nss_chan_msgs_total{channel="foo",server="http://localhost:8222"} 2

# HELP nss_chan_subs_last_sent Last message sent
# TYPE nss_chan_subs_last_sent gauge
nss_chan_subs_last_sent{channel="foo",client_id="16073",server="http://localhost:8222"} 2

# HELP nss_chan_subs_max_inflight Max in flight message count
# TYPE nss_chan_subs_max_inflight gauge
nss_chan_subs_max_inflight{channel="foo",client_id="16073",server="http://localhost:8222"} 1024

# HELP nss_chan_subs_pending_count Pending message count
# TYPE nss_chan_subs_pending_count gauge
nss_chan_subs_pending_count{channel="foo",client_id="16073",server="http://localhost:8222"} 0

# HELP nss_server_bytes_total Total of bytes
# TYPE nss_server_bytes_total counter
nss_server_bytes_total{server="http://localhost:8222"} 48

# HELP nss_server_channels Total channels
# TYPE nss_server_channels counter
nss_server_channels{server="http://localhost:8222"} 1

# HELP nss_server_clients Total clients
# TYPE nss_server_clients counter
nss_server_clients{server="http://localhost:8222"} 1

# HELP nss_server_msgs_total Total of messages
# TYPE nss_server_msgs_total counter
nss_server_msgs_total{server="http://localhost:8222"} 2

# HELP nss_server_subscriptions Total subscriptions
# TYPE nss_server_subscriptions counter
nss_server_subscriptions{server="http://localhost:8222"} 1
```

### CODE REVIEW

File: prometheus_nats_exporter.go [hunk 1]
```diff
@@ -105,6 +105,8 @@ func main() {
 	flag.BoolVar(&opts.GetConnz, "connz", false, "Get connection metrics.")
 	flag.BoolVar(&opts.GetRoutez, "routez", false, "Get route metrics.")
 	flag.BoolVar(&opts.GetSubz, "subz", false, "Get subscription metrics.")
+	flag.BoolVar(&opts.GetStreamingChannelz, "channelz", false, "Get channel metrics.")
```
Comment (ColinSullivan1) [inline prometheus_nats_exporter.go]: "Get streaming channel metrics." ?

File: collector/streaming.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+func (nc *channelsCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- nc.chanBytesTotal
+	ch <- nc.chanMsgsTotal
+	ch <- nc.chanLastSeq
+	ch <- nc.subsLastSent
+	ch <- nc.subsPendingCount
+	ch <- nc.subsMaxInFlight
+}
+
+func (nc *channelsCollector) Collect(ch chan<- prometheus.Metric) {
```
Comment (ColinSullivan1) [inline collector/streaming.go]: I think at some point we'll want "clientz", but this is great for now.

Comment (wallyqs) [reply collector/streaming.go]: Had also looked at clientsz, but seems that we could get all the same data from using channelsz, for example these two carry the same information only difference is that it was fetched from different endpoints (clientsz and channelsz respectively):

```sh
nss_client_last_sent{channel="foo",client_id="client-id-0",server="http://127.0.0.1:8222"} 46162
nss_chan_subs_last_sent{channel="foo",client_id="client-id-0",server="http://127.0.0.1:8222"} 46162
```

**Comment by complex64** [comment]
Is it possible that `nss_chan_bytes_total` and `nss_chan_msgs_total` are in fact gauges (and hence break monotonicity) in the case the streaming server removes messages past the retention window?

**Comment by Will2817** [comment]
@complex64 Yes, I believe that got fixed here: https://github.com/nats-io/prometheus-nats-exporter/pull/76

**Pull request merged**